File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.kt
class Solution {
    fun solution(targets: Array<IntArray>): Int {
        var answer: Int = 0
        val sortedTarget = targets.sortedBy { it[1] }
        
        var e = 0
        for(target in sortedTarget) {
            if(e <= target[0]) {
                answer += 1
                e = target[1]
            }
        }
        return answer
    }
}

File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.js
function solution(targets) {
    var answer = 0;
    const sortedTargets = targets.sort((a, b) => a[1] - b[1]);
    let e = 0;
    sortedTargets.forEach(target => {
        if(e <= target[0]) {
            answer++;
            e = target[1]
        }
    })
    return answer;
}